### PR TITLE
chore: move build-man workflow away from shell

### LINKED ISSUE #11717: `src/doc/build-man.sh` is not able to run on platforms missing `sh`
### Problem

From <https://github.com/rust-lang/cargo/pull/11646#issuecomment-1425034308>, people struggle to run `src/doc/build-man.sh` on Windows. This might prevent Windows users from contributing Cargo documentations. We may want to find a way making every contributor feel inclusive without introducing more external tool dependencies.

### Proposed Solution

Two options off the top of my head:

* **EASY**: Translate `src/doc/build-man.sh` to Windows batch script or powershell.
* **HARD?**: Introduce some workflow like [`cargo-xtask`](https://github.com/matklad/cargo-xtask) (blocked on nested workspaces feature).

Need to take care of `crates/mdman/build-man.sh` as well.

### Notes

Was reading [this article](https://matklad.github.io/2023/02/12/a-love-letter-to-deno.html) and it reminded me this issue.

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

Turn `src/doc/build-man.sh` into a Cargo binary target.
So is `crates/mdman/build-man.sh`.

### How should we test and review this PR?

Run `cargo build-man`

### Additional information

We should leverage workspace inheritance for dependencies.

Fixes #11717

### CODE REVIEW

File: crates/mdman/src/bin/build-man.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
+use std::fs;
```
Comment (epage) [inline crates/mdman/src/bin/build-man.rs:1]: Is there a reason we put this in `mdman`?

I figured us having specific top-level `xtask-` packages would help raise visibility on what xtasks exist

Comment (weihanglo) [reply crates/mdman/src/bin/build-man.rs]: Good point. I have no idea what I was doing.

BTW, do you feel like we should probably have only crates meant to be published stay in `crates/`, and move others elsewhere?

File: crates/mdman/src/bin/build-man.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process;
+use std::process::Command;
+
+/// Run `build-man` to build the man pages for Cargo.
+/// Run `build-man --self` for mdman example outputs.
+fn main() -> io::Result<()> {
+    if std::env::args().nth(1) == Some("--self".to_owned()) {
```
[the remark this answers is not in the document]
Comment (weihanglo) [reply crates/mdman/src/bin/build-man.rs]: Removed. This is not necessary anymore :)

File: crates/mdman/src/bin/build-man.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                "-o",
+                &format!("{mdman_root}/doc/out"),
+                &format!("{mdman_root}/doc/mdman.md"),
+            ]);
+
+        let status = cmd.status()?;
+        if !status.success() {
+            eprintln!("failed to run command `{cmd:?}`");
+            process::exit(status.code().unwrap_or(1));
+        }
```
[the remark this answers is not in the document]
Comment (weihanglo) [reply crates/mdman/src/bin/build-man.rs]: Have no strong opinion at this moment. I can look back when implementing version bump check. The crate does look nice tho.

**Pull request merged**